src/diffusers/utils/peft_utils.py

```diff
@@ -169,14 +169,18 @@ def get_peft_kwargs(rank_dict, network_alpha_dict, peft_state_dict, is_unet=True
             lora_alpha = set(network_alpha_dict.values()).pop()
 
     # layer names without the Diffusers specific
-    target_modules = list({name.split(".lora")[0] for name in peft_state_dict.keys()})
+    target_modules = list(
+        {name.split(".lora")[0] for name in peft_state_dict.keys()}
+    )  # if "lora_magnitude_vector" not in name
+    use_dora = any("lora_magnitude_vector" in k for k in peft_state_dict)
 
     lora_config_kwargs = {
         "r": r,
         "lora_alpha": lora_alpha,
         "rank_pattern": rank_pattern,
         "alpha_pattern": alpha_pattern,
         "target_modules": target_modules,
+        "use_dora": use_dora,
```

Comment:
I think we'll need to pop this value from the dict otherwise later on we'll have `LoraConfig(**lora_config_kwargs)` and it'll error out for users that have the wrong peft version, even if `use_dora` is set to False no?

Reply:
Nice catch!

Does this work for you?
0b7fd1d5